src/main/ruby/truffleruby/core/match_data.rb

```diff
@@ -153,6 +163,21 @@ def match_length(n)
   def to_s
     self[0]
   end
+
+  private
+
+  def backref_from_arg(index)
+    if Primitive.is_a?(index, String) || Primitive.is_a?(index, Symbol)
+      names_to_backref = Hash[Primitive.regexp_names(self.regexp)]
+      key = index.to_sym
+
+      raise IndexError, "undefined group name reference: #{index}" unless names_to_backref.key?(key)
+
+      return names_to_backref[key].last
```

Comment:
This also does two lookups.
Here it seems trivial to avoid with `array = h[k]; raise unless array` since `nil` is not a valid value in that `Hash[Symbol => Array]`

Reply:
Ah, quite right!  I started by wondering if ruby had a function like go's map fetch to tell you if the key existed and then got hung up checking for existence when it's obvious we were never expecting a nil there so it wasn't necessary.